vmm/src/device_manager/mmio.rs

```diff
@@ -312,10 +313,22 @@ mod tests {
             format!("{}", e),
             "unable to add device to kernel command line: string contains an equals sign"
         );
-        let e = Error::CloneIoeventFd(sys_util::Error::new(0));
-        assert_eq!(format!("{}", e), "failed to clone ioeventfd: Error(0)");
-        let e = Error::CloneIrqFd(sys_util::Error::new(0));
-        assert_eq!(format!("{}", e), "failed to clone irqfd: Error(0)");
+        let e = Error::CloneIoeventFd(io::Error::from_raw_os_error(0));
+        assert_eq!(
+            format!("{}", e),
+            format!(
+                "failed to clone ioeventfd: {:?}",
+                io::Error::from_raw_os_error(0)
+            )
+        );
+        let e = Error::CloneIrqFd(io::Error::from_raw_os_error(0));
+        assert_eq!(
+            format!("{}", e),
+            format!(
+                "failed to clone irqfd: {:?}",
```

Comment:
```suggestion
                "failed to clone irqfd: No error information (os error 0)",
```
and remove the `format!`